packages/starlight/user-components/LinkCard.astro

```diff
@@ -1,20 +1,36 @@
 ---
 import Icon from './Icon.astro';
+import Badge from '../components/Badge.astro';
+import type { Badge as BadgeType } from '../schemas/badge';
+import { BadgeConfigSchema } from '../schemas/badge';
 import type { HTMLAttributes } from 'astro/types';
 
 interface Props extends Omit<HTMLAttributes<'a'>, 'title'> {
 	title: string;
+	badge: BadgeType | undefined;
```

Comment:
```suggestion
	badge?: BadgeUserConfig;
```

This needs to be optional otherwise TypeScript would complain if a `<LinkCard>` is called without a `badge` property.

Regarding `BadgeUserConfig`, you cannot use the `Badge` type exported from `packages/starlight/schemas/badge.ts` as it only contains the object definition and not the string shorthand.

To get a type accepting both, you would need to export a new type in `packages/starlight/schemas/badge.ts`:

```ts
export type BadgeUserConfig = z.input<ReturnType<typeof BadgeConfigSchema>>;
```

Reply:
Ah thank you. I assumed there was something I was missing here and this makes sense

Reply:
Unresolving this to not forget it but `export type BadgeUserConfig = z.input<ReturnType<typeof BadgeConfigSchema>>;` would still need to be added to `packages/starlight/schemas/badge.ts`.

Reply:
Thanks for that, just added it to the latest commit